arcgis-ios-sdk-samples/Maps/Display drawing status/MapViewDrawStatusViewController.swift

```diff
@@ -41,23 +41,26 @@ class MapViewDrawStatusViewController: UIViewController {
         
         //assign the map to mapView
         self.mapView.map = self.map
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
         
         //add observer for drawStatus on mapView
-        //so we can show/hide an indicator when the status change 
-        self.mapView.addObserver(self, forKeyPath: "drawStatus", options: .new, context: nil)
+        //so we can show/hide an indicator when the status change
+        mapView.addObserver(self, forKeyPath: #keyPath(AGSGeoView.drawStatus), options: .initial, context: nil)
```

Comment:
The initial option ensures that the view is updated correctly when it first displays. The new option was unnecessary because we consult the map view directly for the new value.